filebeat/harvester/log.go

```diff
@@ -74,7 +74,7 @@ func (h *Harvester) Harvest() {
 
 				h.SetOffset(0)
 				seeker.Seek(h.getOffset(), os.SEEK_SET)
-				continue
+				return
```

Comment:
It is the harvester that detects that the file was truncated, not prospector. The prospector only sees a finished harvester and opens a new one if the offset (which is 0).